.github/workflows/failureNotifier.yml

```
@@ -0,0 +1,85 @@
+name: Notify on Workflow Failure
+
+on:
+  workflow_run:
+    workflows: ["Process new code merged to main"]
+    types:
+      - completed
+
+permissions:
+  issues: write
+
+jobs:
+  notify-failure:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
+      - name: Fetch Workflow Run Jobs
+        id: fetch-workflow-jobs
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const runId = "${{ github.event.workflow_run.id }}";
+            const jobsData = await github.rest.actions.listJobsForWorkflowRun({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: runId,
+            });
+            return jobsData.data;
+
+      - name: Process Each Failed Job
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const jobs = ${{steps.fetch-workflow-jobs.outputs.result}};
+
+            const headCommit = "${{ github.event.workflow_run.head_commit.id }}";            
+            const prData = await github.rest.repos.listPullRequestsAssociatedWithCommit({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              commit_sha: headCommit,
+            });
+
+            const pr = prData.data[0];
+            const prLink = pr.html_url;
+            const prAuthor = pr.user.login;
+            const prMerger = "${{ github.event.workflow_run.actor.login }}";
+
+            const failureLabel = 'workflow-failure';
+            for (let i = 0; i < jobs.total_count; i++) {
+              if (jobs.jobs[i].conclusion == 'failure') {
+                const jobName = jobs.jobs[i].name;
+                const jobLink = jobs.jobs[i].html_url;
+                const issues = await github.rest.issues.listForRepo({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  labels: failureLabel,
+                  state: 'open'
+                });
+                const existingIssue = issues.data.find(issue => issue.title.includes(jobName));
+                if (!existingIssue) {
+                  const issueTitle = `🔍 Investigation Needed: ${ jobName } Failure due to PR Merge 🔍`;
```

```suggestion
                  const issueTitle = `Investigate workflow job failing on main: ${ jobName }`;
```
I think it's a bit tidier to phrase the title like this. We should maybe also prepend a priority in square brackets, e.g. [HIGH] but I'm unsure if there's an internal process for assigning the level? cc @pecanoro